scripts/Interface/reduction_gui/widgets/toftof/toftof_setup.py

```diff
@@ -0,0 +1,431 @@
+            return not runs.strip() and not comment.strip()
+
+        def _removeTrailingEmptyRows(self):
+            for row in reversed(range(self._numRows())):
+                if self._isRowEmpty(row):
+                    del self.dataRuns[row]
+                else:
+                    break
+
+        def _ensureHasRows(self, numRows):
+            while self._numRows() < numRows:
+                self.dataRuns.append(('', ''))
+
+        def _setCellText(self, row, col, text):
+            self._ensureHasRows(row + 1)
+            (runText, comment) = self.dataRuns[row]
+
+            text = text.strip()
+            if col == 0:
+                runText = text
+            else:
+                comment = text
+
+            self.dataRuns[row] = (runText, comment)
+
+        def _getCellText(self, row, col):
+            return self._getRow(row)[col].strip()
+
+        # reimplemented QAbstractTableModel methods
+
+        headers    = ('Data runs', 'Comment')
+        selectCell = pyqtSignal(int, int)
+
+        def rowCount(self, _ = QModelIndex()):
+            # one additional row for new data
+            return self._numRows() + 1
+
+        def columnCount(self, _ = QModelIndex()):
+            return 2
+
+        def headerData(self, section, orientation, role):
+            if Qt.Horizontal == orientation and Qt.DisplayRole == role:
+                return self.headers[section]
+
+            return None
+
+        def data(self, index, role):
+            if Qt.DisplayRole == role or Qt.EditRole == role:
+                return self._getCellText(index.row(), index.column())
+
+            return None
+
+        def setData(self, index, text, _):
+            row = index.row()
+            col = index.column()
+
+            self._setCellText(row, col, text)
+            self._removeTrailingEmptyRows()
+
+            # signal the attached view
+            self.reset()
+
+            # move selection to the next column or row
+            col = col + 1
+
+            if col >= 2:
+                row = row + 1
+                col = 0
+
+            row = min(row, self.rowCount() - 1)
+            self.selectCell.emit(row, col)
+
+            return True
+
+        def flags(self, _):
+            return Qt.ItemIsEnabled | Qt.ItemIsSelectable | Qt.ItemIsEditable
+
+    # tooltips
+    TIP_prefix  = ''
+    TIP_dataDir = ''
+    TIP_btnDataDir = ''
+
+    TIP_vanRuns = ''
+    TIP_vanCmnt = ''
+
+    TIP_ecRuns = ''
+    TIP_ecFactor = ''
+
+    TIP_binEon = ''
+    TIP_binEstart = ''
+    TIP_binEstep = ''
+    TIP_binEend = ''
+
+    TIP_binQon = ''
+    TIP_binQstart = ''
+    TIP_binQstep = ''
+    TIP_binQend = ''
+
+    TIP_maskDetectors = ''
+
+    TIP_dataRunsView = ''
+
+    TIP_chkSubtractECVan = ''
+
+    TIP_rbtNormaliseNone = ''
+    TIP_rbtNormaliseMonitor = ''
+    TIP_rbtNormaliseTime = ''
+
+    TIP_rbtCorrectTOFNone = ''
+    TIP_rbtCorrectTOFVan = ''
+    TIP_rbtCorrectTOFSample = ''
```

Comment:
If I understood correctly, these tooltips are not used for the moment. Are you planning to at some point?